app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -285,6 +286,21 @@ function uploadTypeFromFileName(fileName) {
   return isAnImage(fileName) ? 'image' : 'attachment';
 }
 
+function isGUID(value) {
+  return /^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-fA-F]{12}$/i.test(value);
```

Comment:
In the last `[0-9a-fA-F]{12}`, the uppercase `A-F` isn't need since the regex is globally case insensitive ;)